src/SidebarBottom.tsx

```diff
@@ -1,34 +1,71 @@
 import { type API, useStorybookState } from "@storybook/manager-api";
+import { styled } from "@storybook/theming";
 import type { API_FilterFunction } from "@storybook/types";
-import React, { useCallback } from "react";
+import React, { useCallback, useEffect } from "react";
 
 import { SidebarToggleButton } from "./components/SidebarToggleButton";
-import { ADDON_ID } from "./constants";
+import { ADDON_ID, ENABLE_FILTER } from "./constants";
 
 const filterNone: API_FilterFunction = () => true;
 const filterWarn: API_FilterFunction = ({ status }) => status?.[ADDON_ID]?.status === "warn";
+const filterError: API_FilterFunction = ({ status }) => status?.[ADDON_ID]?.status === "error";
+const filterBoth: API_FilterFunction = ({ status }) =>
+  status?.[ADDON_ID]?.status === "warn" || status?.[ADDON_ID]?.status === "error";
+
+const Wrapper = styled.div({
+  display: "flex",
+  gap: 5,
+});
 
 interface SidebarBottomProps {
   api: API;
 }
 
-export const ENABLE_FILTER = "enableFilter";
-
 export const SidebarBottom = ({ api }: SidebarBottomProps) => {
-  const onEnable = useCallback(() => {
-    api.experimental_setFilter(ADDON_ID, filterWarn);
-    // Used internally to trigger next step in guided tour
-    api.emit(ENABLE_FILTER, ADDON_ID, filterWarn);
-  }, [api]);
-  const onDisable = useCallback(() => api.experimental_setFilter(ADDON_ID, filterNone), [api]);
+  const [showWarnings, setShowWarnings] = React.useState(false);
+  const [showErrors, setShowErrors] = React.useState(false);
 
   const { status } = useStorybookState();
   const warnings = Object.values(status).filter((value) => value[ADDON_ID]?.status === "warn");
-  if (!warnings.length) return null;
+  const errors = Object.values(status).filter((value) => value[ADDON_ID]?.status === "error");
+  const hasWarnings = warnings.length > 0;
+  const hasErrors = errors.length > 0;
+
+  const toggleWarnings = useCallback(() => setShowWarnings((shown) => !shown), []);
+  const toggleErrors = useCallback(() => setShowErrors((shown) => !shown), []);
+
+  useEffect(() => {
+    let filter = filterNone;
+    if (hasWarnings && showWarnings) filter = filterWarn;
+    if (hasErrors && showErrors) filter = filter === filterWarn ? filterBoth : filterError;
```

Comment:
I personally prefer self invocing functions or a `getFilter` function instead of reassignments in line 39 and 40, which makes the code more readable and interrelated